Library/Homebrew/test/rubocops/lines_spec.rb

```diff
@@ -279,6 +279,20 @@ def options
   end
 end
 
+describe RuboCop::Cop::FormulaAudit::Lines do
+  subject(:cop) { described_class.new }
+
+  it "reports an offense when using depends_on \"mpich\"" do
+    expect_offense(<<~RUBY)
+      class Foo < Formula
+        url 'https://brew.sh/foo-1.0.tgz'
+        depends_on "mpich"
+        ^^^^^^^^^^^^^^^^^^ depends_on \"mpich\" is deprecated. Usage should be depends_on \"open-mpi\".
```

Comment:
`brew tests --verbose --debug --only=rubocops/lines` shows:

```
  1) RuboCop::Cop::FormulaAudit::Lines reports an offense when using depends_on "mpich"
     Failure/Error:
       expect_offense(<<~RUBY)
         class Foo < Formula
           url 'https://brew.sh/foo-1.0.tgz'
           depends_on "mpich"
           ^^^^^^^^^^^^^^^^^^ depends_on \"mpich\" is deprecated. Usage should be depends_on \"open-mpi\".
         end
       RUBY

       expected: "class Foo < Formula\n  url 'https://brew.sh/foo-1.0.tgz'\n  depends_on \"mpich\"\n  ^^^^^^^^^^^^^^^^^^ depends_on \"mpich\" is deprecated. Usage should be depends_on \"open-mpi\".\nend\n"
            got: "class Foo < Formula\n  url 'https://brew.sh/foo-1.0.tgz'\n  depends_on \"mpich\"\nend\n"

       (compared using ==)

       Diff:
       @@ -1,6 +1,5 @@
        class Foo < Formula
          url 'https://brew.sh/foo-1.0.tgz'
          depends_on "mpich"
       -  ^^^^^^^^^^^^^^^^^^ depends_on "mpich" is deprecated. Usage should be depends_on "open-mpi".
        end

     # ./vendor/bundle/ruby/2.3.0/gems/rubocop-0.71.0/lib/rubocop/rspec/expect_offense.rb:99:in `expect_offense'
     # ./test/rubocops/lines_spec.rb:286:in `block (2 levels) in <top (required)>'
     # ./test/spec_helper.rb:163:in `block (2 levels) in <top (required)>'
     # ./vendor/bundle/ruby/2.3.0/gems/rspec-retry-0.6.1/lib/rspec/retry.rb:123:in `block in run'
     # ./vendor/bundle/ruby/2.3.0/gems/rspec-retry-0.6.1/lib/rspec/retry.rb:110:in `loop'
     # ./vendor/bundle/ruby/2.3.0/gems/rspec-retry-0.6.1/lib/rspec/retry.rb:110:in `run'
     # ./vendor/bundle/ruby/2.3.0/gems/rspec-retry-0.6.1/lib/rspec_ext/rspec_ext.rb:12:in `run_with_retry'
     # ./vendor/bundle/ruby/2.3.0/gems/rspec-retry-0.6.1/lib/rspec/retry.rb:37:in `block (2 levels) in setup'
     # ./vendor/bundle/ruby/2.3.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block (2 levels) in <top (required)>'
```

Testing locally after modifying a formula (e.g., `opencoarrays`) to depend on MPICH shows that the `--fix` flag works and the message is:

```
opencoarrays:
  * C: 17: col 3: FormulaAudit/MpiChoice: 'depends_on "mpich"' should be replaced with 'depends_on "open-mpi"'.
```

I've poked around other cops and the API documentation, but, currently, my ruby-foo is not quite strong enough to understand why other tests in `lines.rb` pass, yet this one doesn't. Any help/advice from @reitermarkus @MikeMcQuaid and anyone else would be much appreciated.